pilot/pkg/model/jwks_resolver.go

```diff
@@ -242,16 +245,12 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 	if jwksURI == "" {
 		// Fetch the jwks URI if it is not hardcoded on config.
 		jwksURI, err = r.resolveJwksURIUsingOpenID(issuer)
```

Comment:
I am a bit confused. I thought this PR was to make the resolve async, but this seems to be synchronous?